src/tool/stdio_mcp.rs

```diff
@@ -0,0 +1,178 @@
+//! # Model Context Protocol Tools
+//!
+//! This module external tools that can connect with STDIOMCP Servers.
+//!
+//!
+//!
+
+use crate::tool::{Tool, ToolBox, ToolError};
+use anyhow::Result as AnyhowResult;
+use async_trait::async_trait;
+use log::debug;
+use rmcp::{
+    model::CallToolRequestParam,
+    service::RunningService,
+    transport::{ConfigureCommandExt, TokioChildProcess},
+    RoleClient, ServiceExt,
+};
+use serde_json::Value;
+use tokio::process::Command;
+
+pub struct StdIoMcp {
```

Comment:
It is preferable that `MCPToolBox` will be single struct. It can be configured with enum and generics to decide on transport layer